source/summit/setup.d

```diff
@@ -51,9 +51,17 @@ public final class SetupApplication : Application
      * Real index page
      */
     @path("setup") @method(HTTPMethod.GET)
-    void setupIndex() @safe
+    void setupIndex(FormError _error = FormError.init, string instanceURI = null,
+            string description = null, string username = null, string emailAddress = null) @safe
     {
-        render!"setup/index.dt";
+        /* Throw internal error if not from field validation */
+        if (!_error.error.empty && _error.field.empty)
```

Comment:
Could I ask for one minor change here? Combine these into an `enforceHTTP` which will vastly simplify the codeflow. Other than that i love it!!

Reply:
Thanks for the suggestion :+1:  7e1fef1